examples/bri-3/src/bri/communication/agents/messaging.agent.spec.ts

```diff
@@ -0,0 +1,132 @@
+    // Arrange
+    const rawMessage =
+      '{ "id": "0a3dd67c-c031-4b50-95df-0bc5fc1c78b5", "from": "71302cec-0a38-469a-a4e5-f58bdfc4ab32", "to": "76cdd901-d87d-4c87-b572-155afe45c128", "content": 123, "signature": "xyz", "type": 0}';
+
+    // Act
+    const messagingAgent = new MessagingAgent(null, null, null);
+    const [, validationErrors] =
+      messagingAgent.validateBpiMessageFormat(rawMessage);
+
+    // Assert
+    expect(validationErrors.length).toBe(1);
+    expect(validationErrors[0]).toEqual('content: 123 is not valid JSON.');
+  });
+
+  it('Should return error when validating correct JSON raw message with signature field empty', () => {
+    // Arrange
+    const rawMessage =
+      '{ "id": "0a3dd67c-c031-4b50-95df-0bc5fc1c78b5", "from": "71302cec-0a38-469a-a4e5-f58bdfc4ab32", "to": "76cdd901-d87d-4c87-b572-155afe45c128", "content": { "testProp":"testValue" }, "signature": "", "type": 0}';
+
+    // Act
+    const messagingAgent = new MessagingAgent(null, null, null);
+    const [, validationErrors] =
+      messagingAgent.validateBpiMessageFormat(rawMessage);
+
+    // Assert
+    expect(validationErrors.length).toBe(1);
+    expect(validationErrors[0]).toEqual('signature is empty');
+  });
+
+  it('Should return error when validating correct JSON raw message with message type not being info', () => {
+    // Arrange
+    const rawMessage =
+      '{ "id": "0a3dd67c-c031-4b50-95df-0bc5fc1c78b5", "from": "71302cec-0a38-469a-a4e5-f58bdfc4ab32", "to": "76cdd901-d87d-4c87-b572-155afe45c128", "content": { "testProp":"testValue" }, "signature": "xyz", "type": 1}';
+
+    // Act
+    const messagingAgent = new MessagingAgent(null, null, null);
+    const [, validationErrors] =
+      messagingAgent.validateBpiMessageFormat(rawMessage);
+
+    // Assert
+    expect(validationErrors.length).toBe(1);
+    expect(validationErrors[0]).toEqual('type is unknown 1');
+  });
+
+  it('Should return no errors and create message dto when validating correct JSON raw message', () => {
+    // Arrange
+    const rawMessage =
+      '{ "id": "0a3dd67c-c031-4b50-95df-0bc5fc1c78b5", "from": "71302cec-0a38-469a-a4e5-f58bdfc4ab32", "to": "76cdd901-d87d-4c87-b572-155afe45c128", "content": { "testProp":"testValue" }, "signature": "xyz", "type": 0}';
+
+    // Act
+    const messagingAgent = new MessagingAgent(null, null, null);
+    const [resultDto, validationErrors] =
+      messagingAgent.validateBpiMessageFormat(rawMessage);
+
+    // Assert
+    expect(validationErrors.length).toBe(0);
+    expect(resultDto).toBeDefined();
+    expect(resultDto.id).toEqual('0a3dd67c-c031-4b50-95df-0bc5fc1c78b5');
+    expect(resultDto.from).toEqual('71302cec-0a38-469a-a4e5-f58bdfc4ab32');
+    expect(resultDto.to).toEqual('76cdd901-d87d-4c87-b572-155afe45c128');
+    // TODO: Find a fix for the escape character
+    // expect(resultDto.content).toEqual("{ \"testProp\": \"testValue\" }");
```

Comment:
This should work:     expect(resultDto.content).toEqual({ testProp: 'testValue' });

Reply:
Thanks, missed to come back to this TODO. Fixed.